src/lib.rs

```diff
@@ -169,6 +169,44 @@ pub struct IncomingRequests<'a> {
     server: &'a Server,
 }
 
+/// Control buffering in a data stream.
+#[derive(Debug, Copy, Clone)]
+pub enum BufferingMode {
+    /// Buffer the messages. This is the default.
+    Buffered,
+    /// Do not buffer the messages, this may incur a performance cost.
+    Unbuffered,
+}
+
+impl Default for BufferingMode {
+    fn default() -> Self {
+        BufferingMode::Buffered
+    }
+}
+
+/// Advanced server settings.
+/// In order to retain the ability to add options later while preserving the
+/// API, this is an "opaque" struct, to be manipulated through the `new` and
+/// `with_*` configuration methods.
+#[derive(Default, Debug, Clone)]
+pub struct ServerConfigAdvanced {
+    /// Control buffering on the server->client path. The default value is
+    /// `BufferingMode::Buffered`.
+    writer_buffering: BufferingMode,
+}
+
+impl ServerConfigAdvanced {
+    pub fn new() -> Self {
+        Self::default()
```

Comment:
To be honest, I'm not certain providing a default impl of `ServerConfigAdvanced` is a great idea, maybe I should drop that and explicitly instantiate the configuration in the `new` method.
Because the `Default` impl is public-facing, so if we commit to that, we will need to maintain a `Default` impl on that structure in the future.
That said, we do have to provide default values anyway (wether in the default impl or in the `new` method), so I don't feel strongly about this either way. But if the idea is to provide some opaque structure with a builder pattern, I guess the less we expose API-wide, the better?

Reply:
I don't know if providing a `Defaul` implementation is any more or less public than an opaque `new()`, in that we would need to preserve any behavioural expectations anyway - I would like a builder pattern configuration mechanism anyway, but it doesn't need to be at the expense of `Default`.